src/clib/grackle.h

```diff
@@ -16,6 +16,7 @@
 
 #include "grackle_types.h"
 #include "grackle_chemistry_data.h"
+#include <cstddef>
```

Comment:
I suspect that the following change should fix the CircleCi problems:

```suggestion
#include <stddef.h>
```

For context, ``<cstddef>`` is a special header in the C++ standard library that can be used in place of ``<stddef.h>`` (which is defined in the C standard library). It guarantees that all of the symbols within ``<stddef.h>`` are imported into the ``std`` namespace. Since the compiler treats Grackle's source code as C, the codebase is compiled without access to the C++ standard library, and consequently doesn't know of a header called ``<cstddef>``.